### DISCUSSION TITLE
Using MSBuild on the command line and building to any platform/target possible?

### DISCUSSION THREAD

**Discussion by vurvdev** [dotnet/msbuild#6896] [Q&A] [answered]
Is it possible to use MSBuild on the command line for a ``.sln`` without specifying the platform / configurations (``/p:Configuration=Debug`` or ``/p:Platform="Any CPU"`` for example)?

I am trying to automate this and am going through many different solutions which all have different configurations, so I can't just specify one as many projects won't have it and will lead to this error:
```
The specified solution configuration "...|..." is invalid.
Please specify a valid solution configuration using the Configuration and Platform properties ...
```

Basically, if a solution has many targets such as Release|x64, Debug|Any CPU then I just want MSBuild to pick the first one applicable to build without me specifying it.

**Comment by rainersigwald** [comment, marked as answer]
You don't have to specify a configuration--MSBuild will pick the first one defined in the solution.
